src/main/java/org/opentripplanner/middleware/triptracker/ManageLegTraversal.java

```diff
@@ -0,0 +1,273 @@
+package org.opentripplanner.middleware.triptracker;
+
+import com.grum.geocalc.Coordinate;
+import com.grum.geocalc.EarthCalc;
+import com.grum.geocalc.Point;
+import io.leonard.PolylineUtils;
+import io.leonard.Position;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.Leg;
+import org.opentripplanner.middleware.utils.Coordinates;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+public class ManageLegTraversal {
+
+    private ManageLegTraversal() {
+    }
+
+    /**
+     * Define the shortest distance to any position within a trip by checking the traveler's position against all
+     * positions within all legs.
+     *
+     * @param currentCoordinates Traveler's current position.
+     * @param itinerary Trip itinerary containing all trip legs.
+     * @return The shortest distance to any trip position.
+     */
+    public static double getDistanceFromNearestTripPosition(Coordinates currentCoordinates, Itinerary itinerary) {
+        double shortestDistance = Double.MAX_VALUE;
+        if (canUseTripLegs(itinerary)) {
+            for (Leg leg : itinerary.legs) {
```

Comment:
This is interesting. Maybe we need this method, but just for a leg and for the transition points between legs, instead of the entire itinerary.
My reasoning is that once someone engages in one leg or a transition point between two legs, they are not (should not be) going back to a previous leg.